crates/payload/basic/src/lib.rs

```diff
@@ -532,8 +531,12 @@ where
             };
         }
 
-        let fut =
-            ResolveBestPayload { best_payload, maybe_better, empty_payload, wait_for_pending };
+        let fut = ResolveBestPayload {
+            best_payload,
+            maybe_better,
+            empty_payload,
+            wait_for_pending: kind == PayloadKind::WaitForPending,
+        };
```

Comment:
here we don't need to pass the kind as argument instead we can control it based on whether we set the empty_payload
if `Kind.is_earliest` then we can keep this as none so the ResolveBestPayload will only race the two real payloads against each other

Reply:
rn if payload job is still in progress `ResolveBestPayload` would always return the `best_payload`, so we need to somehow make it poll the pending one until completeion I think?

Reply:
imo this is fine, because you still want to return as fast as possible and if we already have one we should probably return that asap, the pending could even be the same or is not guaranteed to be better

Reply:
updated, though I think there's an edge case when pending job fails leaving us with nothing to return

but given this is only used in --dev mode for now I think it should be OK